### PR TITLE
Add structopt v0.1.0

### PR DESCRIPTION
**Pull request by p-ranav**
Specify library name and version:  **structopt/0.1.0**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/structopt/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+        "clap", "visit-struct-library", "magic-enum")
+    no_copy_source = True
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def configure(self):
+        if self.settings.compiler.cppstd:
+            tools.check_min_cppstd(self, 17)
```
Comment (uilianries) [inline recipes/structopt/all/conanfile.py:25]:
```suggestion
        if self.settings.compiler.cppstd:
            tools.check_min_cppstd(self, "17")
        else:
            self.output.warn("%s recipe lacks information about the %s compiler"
                             " standard version support" % (self.name, compiler))
        
        minimal_version = {
            "Visual Studio": "16",
            "gcc": "7.3",
            "clang": "6",
            "apple-clang": "10.0"
        }

        if compiler not in minimal_version:            
            self.output.info("%s requires a compiler that supports at least"
                             " C++17" % self.name)
            return

        if compiler_version < minimal_version[compiler]:
            raise ConanInvalidConfiguration("%s requires a compiler that supports"
                                            " at least C++17. %s %s is not" 
                                            " supported." % (self.name, compiler, Version(self.settings.compiler.version)))
```

checking only cppstd is not enough.

File: recipes/structopt/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    def package(self):
+        self.copy(pattern="LICENSE", src=self._source_subfolder, dst="licenses")
+        self.copy(pattern="*.h", src=os.path.join(self._source_subfolder, "include"), dst="include")
+        self.copy(pattern="*.hpp", src=os.path.join(self._source_subfolder, "include"), dst="include")
+
+    def package_id(self):
+        self.info.header_only()
+
+    def package_info(self):
+        pass
```
Comment (uilianries) [inline recipes/structopt/all/conanfile.py:40]:
```suggestion
```

Not required

File: recipes/structopt/all/test_package/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,9 @@
+cmake_minimum_required(VERSION 2.8.11)
```
Comment (uilianries) [inline recipes/structopt/all/test_package/CMakeLists.txt:1]:
```suggestion
cmake_minimum_required(VERSION 3.1)
```

CXX_STANDARD is CMake 3

File: recipes/structopt/all/conanfile.py [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+        compiler = str(self.settings.compiler)
+        version = tools.Version(self.settings.compiler.version)
+        if compiler == "Visual Studio" and version >= "15":
+            return True
+        if compiler == "gcc" and version >= "9":
+            return True
+        if compiler == "clang" and version >= "5":
+            return True
+        if compiler == "apple-clang" and version >= "10":
+            return True
```
Comment (uilianries) [inline recipes/structopt/all/conanfile.py:34]:
```suggestion
        if compiler == "Visual Studio" and version >= "15":
            return True
        elif compiler == "gcc" and version >= "9":
            return True
        elif compiler == "clang" and version >= "5":
            return True
        elif compiler == "apple-clang" and version >= "10":
            return True
        else:
            self.output.warn("{} recipe lacks information about the {} compiler standard version support".format(self.name, compiler))
```

I would add an information about other compilers. Only the error can cause confusion.

**Review by danimtb — APPROVED** [review]

**Pull request merged**